javascript/stemwords-es6.mjs

```diff
@@ -0,0 +1,129 @@
+import fs from 'fs';
+import readline from 'readline';
+
+const usage = () => {
+    console.log(`usage: stemwords-es6.mjs [-l <language>] -i <input file> -o <output file> [-c <character encoding>] [-h]
+
+The input file consists of a list of words to be stemmed, one per
+line. Words should be in lower case.
+
+If -c is given, the argument is the character encoding of the input
+and output files.  If it is omitted, the UTF-8 encoding is used.
+
+The output file consists of the stemmed words, one per line.
+
+-h displays this help`);
```

Comment:
I've peeled off this change to use a single `console.log()` call as it makes the source more readable.

The rest seems to be rewriting it to be async?

Not entirely convinced that's helpful - at least to me it makes it harder to follow so worse in it's partial role as an example of how to use the API, but maybe if you write a lot of JS async code it's actually easier to read in that form.

It doesn't seem to be measurably faster as async either (it can probably parallelise IO with stemming a bit, but there is presumably extra overhead from the parallelism too).